### PR TITLE
custom-gas-token

### PR DESCRIPTION
**Pull request by cpengilly**
draft PR, merge only after eng approves

### CODE REVIEW

**Review by BlocksOnAChain — APPROVED** [review]
This looks good to me.

File: pages/stack/protocol/features/custom-gas-token.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+*   cannot be rebasing or have a transfer fee
+*   must be transferrable only via a call to the token address itself
+*   must only be able to set allowance via a call to the token address itself
+*   must not have a callback on transfer, and more generally a user must not be able to make a transfer to themselves revert
+
+
+## Next Steps
+
+*   Ready to get started? Read our guide on how to [deploy your custom gas token chain](/builders/chain-operators/management/custom-gas-token).
+*   For bridges or developers building on a custom gas token chain, read our tutorial on [depositing custom gas tokens into a chain](/builders/app-developers/tutorials/standard-bridge-custom-token).
```
Comment (sbvegan) [inline pages/stack/protocol/features/custom-gas-token.mdx:70]: I think that this link should be removed, it seems like there is a different way of bridging with CGT and this tutorial could be misleading

File: pages/builders/chain-operators/management/custom-gas-token.mdx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  *   must be a valid ERC-20 token
+  *   the number of decimals on the token MUST be exactly 18
+  *   the name of the token MUST be less than or equal to 32 bytes
+  *   symbol MUST be less than or equal to 32 bytes.
+  *   must not be yield-bearing
+  *   cannot be rebasing or have a transfer fee
+  *   must be transferrable only via a call to the token address itself
+  *   must only be able to set allowance via a call to the token address itself
+  *   must not have a callback on transfer, and more generally a user must not be able to make a transfer to themselves revert
```
Comment (tynes) [inline pages/builders/chain-operators/management/custom-gas-token.mdx:43]: we should say `and more generally a user must not be able to make a transfer have unexpected side effects`, this captures the revert case and also all funny business the token may try to do

File: pages/builders/chain-operators/management/custom-gas-token.mdx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+  ```
+  function initiateWithdrawal(
+      address _target,
+      uint256 _gasLimit,
+      bytes memory _data
+  ) public payable;
+  ```
+</Steps>
+
```
Comment (tynes) [inline pages/builders/chain-operators/management/custom-gas-token.mdx:131]: We could add some validation commands:

This calls the `WETH` predeploy which should be considered wrapped native asset in the context of custom gas token. It should return the name of the token prefixed by `"Wrapped "`

```bash
cast call --rpc-url $L2_ETH_RPC_URL 0x4200000000000000000000000000000000000006 'name()(string)'
```

This calls the `L1Block` predeploy should return `true`.

```bash
cast call --rpc-url $L2_ETH_RPC_URL 0x4200000000000000000000000000000000000015 'isCustomGasToken()(bool)
```

This calls the L1 `SystemConfig` contract and should return `true`.

```bash
cast call --rpc-url $L1_ETH_RPC_URL <SYSTEM_CONFIG_ADDRESS> 'isCustomGasToken()(bool)
```

File: pages/stack/protocol/features/custom-gas-token.mdx (hunk 1 above)
Comment (K-Ho) [reply pages/stack/protocol/features/custom-gas-token.mdx]: Great catch, this tutorial is unrelated... Name is confusing the tutorial is for bridging ERC20s via the bridge contract

File: pages/builders/chain-operators/management/custom-gas-token.mdx (hunk 2 above)
Comment (K-Ho) [reply pages/builders/chain-operators/management/custom-gas-token.mdx:43]: added

File: pages/builders/chain-operators/management/custom-gas-token.mdx (hunk 3 above)
Comment (K-Ho) [reply pages/builders/chain-operators/management/custom-gas-token.mdx:152]: added

**Review by K-Ho — APPROVED** [review]

**Pull request merged**